jlm/llvm/opt/DeadNodeElimination.hpp

```diff
@@ -7,6 +7,7 @@
 #define JLM_LLVM_OPT_DEADNODEELIMINATION_HPP
 
 #include <jlm/rvsdg/lambda.hpp>
+#include <jlm/rvsdg/Phi.hpp>
```

Comment:
A forward declaration should have been enough here.

Reply:
I prefer not to as it tends to interfere with inheritance hierarchies (which are not visible to the compiler).

e.g.
class X;
fn(const Y&);
fn(x);  // <-- this may be valid when X inherits from Y, but is not visible by forward decl

I think it can lead to ambiguous overload resolution in some cases, but forgot the details.